fat/src/main.rs

```diff
@@ -79,16 +79,25 @@ async fn main() -> Result<()> {
 }
 
 async fn run(config: Config, db: DB, shutdown: Controller<String>) -> Result<()> {
+	// Fat client requires P2P to function properly, so we should exit if P2P is disabled
+	if matches!(config.network_mode, NetworkMode::RPCOnly) {
+		return Err(eyre!("Fat client cannot run in RPC-only mode. Please enable P2P by setting network_mode to 'Both' or 'P2POnly'."));
```

Comment:
I think we can replace this with `bail!("Fat client cannot run in RPC-only mode. Please enable P2P by setting network_mode to 'Both' or 'P2POnly'.")`